zero_bin/common/src/pre_checks.rs

```diff
@@ -0,0 +1,23 @@
+use proof_gen::proof_types::GeneratedBlockProof;
+
+pub fn check_previous_proof_and_checkpoint(
+    checkpoint_block_number: u64,
+    previous_proof: &Option<GeneratedBlockProof>,
+    start: u64,
+) -> anyhow::Result<()> {
+    if let Some(proof) = previous_proof {
+        if proof.b_height + 1 != start {
+            return Err(anyhow::Error::msg(format!(
+                "Found previous b_height {} whereas range start is {}",
```

Comment:
I'd make the error message more explicit, we don't know where these values are coming from

Reply:
you would like to include more details like rpc_type, details of proof etc?